client/asset/dcr/dcr.go

```diff
@@ -4527,3 +4529,124 @@ func float64PtrStr(v *float64) string {
 	}
 	return strconv.FormatFloat(*v, 'f', 8, 64)
 }
+
+func (dcr *ExchangeWallet) ConfirmRedemption(coinID dex.Bytes, redemption *asset.Redemption) (*asset.ConfirmRedemptionStatus, error) {
+	txHash, _, err := decodeCoinID(coinID)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := dcr.wallet.GetTransaction(dcr.ctx, txHash)
+	if err == nil {
+		spew.Dump(tx)
+		return &asset.ConfirmRedemptionStatus{
+			Confs:  uint64(tx.Confirmations),
+			Req:    reqiredRedeemConfirms,
+			CoinID: coinID,
+		}, nil
+	}
```

Comment:
It does not seem like dcr spv will ever let go of a bad transaction. I have one in my testnet node that I planted, and now am stuck here as it will always find the fake transaction even though it is spending a spent contract.

So, for dcr, I'm not sure we can ever lose a transaction that is not mined. It seems as if there could be the opposite problem and we hang on to an impossible transaction forever.

We could check if the contract is spent yet first.

But I wonder if this should either check the redeem's fee vs the current fee and/or how long the transaction has been sitting and we `abandontransaction` if it looks like it is being ignored by miners and send a new one.

Reply:
Adding a map for dcr that will abandon transactions if they were first seen two hours ago and still have no confirmations.